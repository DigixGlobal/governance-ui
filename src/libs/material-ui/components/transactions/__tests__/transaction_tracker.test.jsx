import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TransactionTracker from '../transaction_tracker';

jest.mock('react-redux', () => ({
  connect: () => component => component,
}));

const networks = [{
  id: 1,
  explorerTransactionPrefix: 'testPrefix',
  explorerBlockPrefix: 'blockTestPrefix',
}, {
  id: 2,
}];

const web3Redux = {
  web3: () => ({
    eth: {
      transaction: () => jest.fn,
    },
    networkId: 1,
  }),
};

describe('<TransactionTracker />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <TransactionTracker
        networks={networks}
        txHash="0x123456"
        networkId="1"
        broadcast={new Date()}
        web3Redux={web3Redux}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
