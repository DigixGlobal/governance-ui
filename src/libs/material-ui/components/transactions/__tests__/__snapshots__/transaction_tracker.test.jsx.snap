// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionTracker /> renders correctly 1`] = `
<div>
  <Table
    as="table"
  >
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
        >
          Transaction
        </TableCell>
        <TableCell
          as="td"
        >
          <a
            href="testPrefix0x123456"
            rel="noopener noreferrer"
            target="_blank"
          >
            0x123456
          </a>
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
        >
          Status
        </TableCell>
        <TableCell
          as="td"
        >
          <Label
            as="a"
            color="orange"
            href="testPrefix0x123456"
            rel="noopener noreferrer"
            target="_blank"
          >
            <Icon
              as="i"
              loading={true}
              name="circle notched"
            />
            Pending
          </Label>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</div>
`;
